### ISSUE TITLE
x_create_picture_with_pictfmt_and_pixmap fails, root window turns black, no shadows

### ISSUE THREAD

**Issue by SolraBizna** [yshui/picom#459]
### Platform

Debian Bullseye, aarch64, on a Pinebook Pro.

### GPU, drivers, and screen setup

Rockchip RK3399 SOC, Mali-T864 GPU inside.

Not sure what my Xorg driver version is, but Mesa is 20.1.2, built from source.

Output of `glxinfo -B`:

```
name of display: :1
display: :1  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Panfrost (0xffffffff)
    Device: Mali T860 (Panfrost) (0xffffffff)
    Version: 20.1.2
    Accelerated: yes
    Video memory: 3794MB
    Unified memory: yes
    Preferred profile: compat (0x2)
    Max core profile version: 0.0
    Max compat profile version: 2.1
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 2.0
OpenGL vendor string: Panfrost
OpenGL renderer string: Mali T860 (Panfrost)
OpenGL version string: 2.1 Mesa 20.1.2
OpenGL shading language version string: 1.20

OpenGL ES profile version string: OpenGL ES 2.0 Mesa 20.1.2
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 1.0.16
```

### Environment

My desktop environment is an uncooperative Sawfish.

### picom version

Built from source from current git

Output of `picom --diagnostics`:

```
**Version:** vgit-fd6ff

### Extensions:

* Shape: Yes
* XRandR: Yes
* Present: Present

### Misc:

* Use Overlay: Yes* Config file used: None

### Drivers (inaccurate):

modesetting  
```

### Configuration:

Problem occurs with a blank configuration file.

### Steps of reproduction

1. Just run `picom`. 

### Expected behavior

`picom` redirects all of my windows, they get redrawn, and then I have whatever visual features I requested. (Shadows, transparency...)

### Current Behavior

`picom` redirects all of my windows, they get redrawn, and then my root window becomes black. Shadows don't appear, no matter what I do, but if I configure windows to be transparent, that works.

This error message is the only output:

```
[ 08/03/2020 04:13:02.784 x_create_picture_with_pictfmt_and_pixmap ERROR ] failed to create picture (X error 8 MATCH request 139 minor 4 serial 1990)
```

Killing `picom` restores my root window, and returns me to a familiar, prehistoric world of watching xterms slowly redraw every time I switch workspaces.

### Stack trace

Backtrace at the point the message is written (using GLX backend and no config file):

```
#0  x_create_picture_with_pictfmt_and_pixmap (c=0xaaaaaab2cd40, 
    pictfmt=0xaaaaaac7a304, pixmap=29360131, valuemask=1, attr=0xfffffffff3e0)
    at ../src/x.c:209
#1  0x0000aaaaaaacdf74 in x_create_picture_with_visual_and_pixmap (
    c=0xaaaaaab2cd40, visual=33, pixmap=29360131, valuemask=1, 
    attr=0xfffffffff3e0) at ../src/x.c:220
#2  0x0000aaaaaaad45e4 in get_root_tile (ps=0xaaaaaab3ad10)
    at ../src/render.c:487
#3  0x0000aaaaaaad46fc in paint_root (ps=0xaaaaaab3ad10, 
    reg_paint=0xfffffffff4d0) at ../src/render.c:521
#4  0x0000aaaaaaad58ac in paint_all (ps=0xaaaaaab3ad10, t=0xaaaaaac7f5c0, 
    ignore_damage=false) at ../src/render.c:844
#5  0x0000aaaaaaaba0a0 in _draw_callback (loop=0xfffff7f006a8, 
    ps=0xaaaaaab3ad10, revents=8192) at ../src/picom.c:1445
#6  0x0000aaaaaaab9f5c in _draw_callback (loop=0xfffff7f006a8, 
    ps=0xaaaaaab3ad10, revents=8192) at ../src/picom.c:1426
#7  0x0000aaaaaaaba1ac in draw_callback (loop=0xfffff7f006a8, 
    w=0xaaaaaab3add0, revents=8192) at ../src/picom.c:1467
#8  0x0000fffff7ee695c in ev_invoke_pending ()
   from /lib/aarch64-linux-gnu/libev.so.4
#9  0x0000fffff7eea198 in ev_run () from /lib/aarch64-linux-gnu/libev.so.4
#10 0x0000aaaaaaabca98 in session_run (ps=0xaaaaaab3ad10)
    at ../src/picom.c:2323
#11 0x0000aaaaaaabcdb4 in main (argc=4, argv=0xfffffffff978)
    at ../src/picom.c:2424
```

### Other details

Backend doesn't matter, tried both glx and xrender backends. xrender backend has other problems and slightly worse performance, though I'm inclined to blame my driver for that and not `picom`.

I expect that the problem will turn out to be something very silly, and the fix something trivial.

**Comment by mighty9245** [comment]
how do you set the wallpaper?

**Comment by SolraBizna** [comment]
Using `xsetbg` (something fullscreen or something tiled) or `xsetroot` (solid color or pattern). Same results all four ways.

**Comment by mighty9245** [comment]
try setting it to a random image with `feh`. if you're not familiar with it, just do `feh --bg-center --no-fehbg ...`, where `...` is the path to an actual image.

**Comment by SolraBizna** [comment]
Interesting. If I set the background with `feh` instead of `xsetroot` or `xsetbg`, the error does not manifest. If I have set the background with `feh` and subsequently use `xsetroot` or `xsetbg` to change the background, starting `picom` restores the appearance of the root window to how `feh` left it, and killing `picom` brings back the most recently set background (regardless of source).

Either way, having set a background with `feh` also makes shadows work. o_O
